### PR TITLE
Login

### PR DESCRIPTION
**Pull request by hamzabisevac**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { Routes, Route, Link } from "react-router-dom";
+import Login from "./components/Login";
+import ForgetPassword from "./components/ForgetPassword";
+import SignUp from "./components/SignUp";
+import Home from "./components/Home";
+import Acc from "./components/SignUp/index";
+import feature from "./components/img/feature.png";
+
+function App() {
+  const [count, setCount] = useState(0);
```
Comment (enverbisevac) [inline src/App.tsx:13]: unused state? pls remove it

File: src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import logo from "./logo.svg";
+import "./App.css";
```
Comment (enverbisevac) [inline src/App.tsx:3]: pls use best practice to sort imports

File: src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import Acc from "./components/SignUp/index";
+import feature from "./components/img/feature.png";
+
+function App() {
+  const [count, setCount] = useState(0);
+
+  return (
+    <div
+      className="App"
+      style={{
```
Comment (enverbisevac) [inline src/App.tsx:18]: you should use styled component for this

File: src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        height: "90vh",
+        width: "100%",
+        backgroundImage: `url(${feature})`,
+      }}
+    >
+      <Routes>
+        <Route path="/" element={<Login />} />
+        <Route path="/ForgetPassword" element={<ForgetPassword />} />
+        <Route path="/SignUp" element={<SignUp />} />
+        <Route path="/Home" element={<Home />} />
```
Comment (enverbisevac) [inline src/App.tsx:28]: you must follow best practice for defining urls:
forgot-password
signup
home -> /

File: src/components/ForgetPassword/index.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import styled from "styled-components";
+import { useForm } from "@mantine/form";
+import { PasswordInput, Group, Button, Box } from "@mantine/core";
+import logo from "../../logo.svg";
+function ForgetPassw() {
+  const form = useForm({
+    initialValues: {
+      password: "secret",
+      confirmPassword: "sevret",
```
Comment (enverbisevac) [inline src/components/ForgetPassword/index.tsx:10]: password should not contain default values

File: src/components/Flag/index.tsx [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+            Identify: <TextInput placeholder="Identify:" />
+          </li>
+          {/* <li>
+            Default On Variont: ON{" "}
+            <TextInput placeholder="Default ON Variont:" />{" "}
+          </li>
+          <li>
+            Default OF Variont: OF{" "}
+            <TextInput placeholder="Default OF Variont:" />{" "}
+          </li> */}
```
Comment (farismecinovic) [inline src/components/Flag/index.tsx:27]: remove comments pls

File: src/App.css [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  background-color: #282c34;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+} */
+
```
Comment (darishodzic) [inline src/App.css:20]: pls remove comments

File: src/App.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import SignUp from "./components/SignUp";
+import Home from "./components/Home";
+import ForgotPassword from "./components/ForgotPassword";
+import styled from "styled-components";
+import MainScreen from "./components/PrivateRoute";
+import Flags from "./components/Flags";
+
+function App() {
+  return (
+    <AppLogo>
```
Comment (darishodzic) [inline src/App.tsx:13]: This should be something like AppWrapper

File: src/components/Flag/index.tsx [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+  const [gender, setGender] = React.useState();
+  const handleChange = (e: any) => {
+    const target = e.target;
+    if (target.checked) {
+      setGender(target.value);
+    }
+  };
+  const handleSubmit = (e: any) => {
+    e.preventDefault();
+    console.log(gender);
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:18]: pls remove clg

File: src/components/Flag/index.tsx [hunk 10]
```diff
@@ -0,0 +1,310 @@
+import { Paper, Radio, RadioGroup, TextInput } from "@mantine/core";
+import { formList } from "@mantine/form";
+import React, { useCallback, useState } from "react";
+import styled from "styled-components";
+
+function Flag(props: { flag: string }) {
+  const [first, setFirst] = useState({ on: false, off: false });
+  const [activeFlags, setActiveFlags] = useState<any[]>([]);
+  const [gender, setGender] = React.useState();
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:9]: what you think about 'useState'  instead of 'React.useState'

File: src/components/Flag/index.tsx [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+            </th>
+            <th>Identify</th>
+            <th>Name</th>
+            <th>Description</th>
+            <th>Value</th>
+          </tr>
+
+          <tr
+            style={{
+              background: activeFlags.includes("red-1")
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:51]: please make theme and use them

File: src/components/Flag/index.tsx [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+                <form onSubmit={handleSubmit}>
+                  {" "}
+                  <RadionCenter>
+                    <Radio
+                      value="red-1"
+                      checked={gender === "red-1"}
+                      onChange={handleChange}
+                      onClick={() => {
+                        setActiveFlags(["red-1"]);
+                        console.log(activeFlags);
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:69]: remove clg

File: src/components/Flag/index.tsx [hunk 13]
```diff
@@ -0,0 +1,310 @@
...
+
+              <td>
+                <RadionCenter>
+                  <Radio
+                    value="red-2"
+                    checked={gender === "red-2"}
+                    onChange={handleChange}
+                    onClick={() => {
+                      setActiveFlags(["red-2"]);
+                      console.log(activeFlags);
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:84]: remove clg

File: src/components/Flag/index.tsx [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+              <TextInput placeholder="True" size="xs" />
+            </td>
+            <td>
+              <TextInput placeholder="/" size="xs" />
+            </td>
+          </tr>
+
+          <tr
+            style={{
+              backgroundColor: activeFlags.includes("red-3")
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:106]: use theme and pls dont use inline styles

File: src/components/Flag/index.tsx [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+                  {" "}
+                  <RadionCenter>
+                    <Radio
+                      disabled={first.on}
+                      value="red-3"
+                      checked={gender === "red-3"}
+                      onChange={handleChange}
+                      onClick={() => {
+                        setActiveFlags(["red-3"]);
+                        console.log(activeFlags);
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:125]: remove clg

File: src/components/Flag/index.tsx [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+                <td>
+                  {" "}
+                  <RadionCenter>
+                    <Radio
+                      value="red-4"
+                      checked={gender === "red-4"}
+                      onChange={handleChange}
+                      onClick={() => {
+                        setActiveFlags(["red-4"]);
+                        console.log(activeFlags);
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:139]: remove clg

File: src/components/Flag/index.tsx [hunk 17]
```diff
@@ -0,0 +1,310 @@
...
+              <TextInput placeholder="/" size="xs" />
+            </td>
+
+            <td>
+              <TextInput placeholder="True" size="xs" />
+            </td>
+          </tr>
+          <tr
+            style={{
+              backgroundColor: activeFlags.includes("red-5")
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:163]: use theme and pls dont use inline css

File: src/components/Flag/index.tsx [hunk 18]
```diff
@@ -0,0 +1,310 @@
...
+              <td>
+                {" "}
+                <RadionCenter>
+                  <Radio
+                    value="red-5"
+                    checked={gender === "red-5"}
+                    onChange={handleChange}
+                    onClick={() => {
+                      setActiveFlags(["red-5"]);
+                      console.log(activeFlags);
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:180]: remove clg

File: src/components/Flag/index.tsx [hunk 19]
```diff
@@ -0,0 +1,310 @@
...
+    gap: 20px;
+  }
+  li,
+  input {
+    border: none;
+    background-color: #effbff;
+    font-size: 16px;
+  }
+  li:nth-child(1) {
+    border-bottom: 1px solid #0278d5;
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:251]: use theme pls

File: src/components/Flag/index.tsx [hunk 20]
```diff
@@ -0,0 +1,310 @@
...
+    width: 200px;
+  }
+  li:nth-child(2) {
+    border-bottom: 1px solid #0278d5;
```
Comment (darishodzic) [inline src/components/Flag/index.tsx:255]: use theme pls

File: src/components/Flags/index.tsx [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+  width: 300px;
+`;
+const WrapperFlags = styled.div`
+  color: black;
+  font-size: 16px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  border-top: 1px solid #d9dae5;
+  border-bottom: 1px solid #d9dae5;
```
Comment (darishodzic) [inline src/components/Flags/index.tsx:78]: theme

File: src/components/Flags/index.tsx [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+  justify-content: center;
+  align-items: center;
+  margin: 0 !important;
+  pointer-events: none;
+`;
+
+const Background = styled.div`
+  height: 74.1%;
+  padding: 1px;
+  background-color: #effbff;
```
Comment (darishodzic) [inline src/components/Flags/index.tsx:101]: theme

File: src/components/ForgotPassword/index.tsx [hunk 23]
```diff
@@ -0,0 +1,62 @@
...
+    },
+
+    validate: {
+      email: (value) => (/^\S+@\S+$/.test(value) ? null : "Invalid email"),
+    },
+  });
+
+  return (
+    <div
+      style={{
```
Comment (darishodzic) [inline src/components/ForgotPassword/index.tsx:20]: pls dont use inline css

File: src/components/Login/index.tsx [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+    },
+  });
+
+  return (
+    <div
+      style={{
+        height: "100vh",
+        width: "100%",
+        backgroundImage: `url(${feature})`,
+      }}
```
Comment (darishodzic) [inline src/components/Login/index.tsx:51]: pls dont use inline css

File: src/components/Login/index.tsx [hunk 25]
```diff
@@ -0,0 +1,136 @@
...
+        </Box>
+      </Wrapper>
+    </div>
+  );
+}
+
+const Wrapper = styled.div`
+  padding: 40px;
+  -webkit-box-shadow: 0px 2px 15px 5px #dedede;
+  box-shadow: 0px 2px 15px 5px #dedede;
```
Comment (darishodzic) [inline src/components/Login/index.tsx:100]: theme

File: src/components/Login/index.tsx [hunk 26]
```diff
@@ -0,0 +1,136 @@
...
+const Acc = styled.div`
+  display: flex;
+  gap: 15px;
+  align-items: center;
+  text-align: center;
+  justify-content: center;
+  padding: 20px;
+`;
+const SignUpp = styled.div`
+  color: #0278d5;
```
Comment (darishodzic) [inline src/components/Login/index.tsx:117]: theme

File: src/components/Login/index.tsx [hunk 27]
```diff
@@ -0,0 +1,136 @@
...
+const Email = styled.div`
+  width: 45px;
+  margin-bottom: 12px;
+`;
+const PassForg = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 20px;
+`;
+const Password = styled.div``;
```
Comment (darishodzic) [inline src/components/Login/index.tsx:130]: empty?

File: src/components/Navigation/index.tsx [hunk 28]
```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+import { Navbar, ScrollArea } from "@mantine/core";
+
+function Navigation() {
+  const navigate = useNavigate();
+  const [color, setColor] = useState<any[]>([]);
```
Comment (darishodzic) [inline src/components/Navigation/index.tsx:8]: type?

File: src/components/Navigation/index.tsx [hunk 29]
```diff
@@ -0,0 +1,108 @@
...
+      <Text
+        onChange={handleChange}
+        onClick={() => {
+          navigate("/Projects");
+          setColor(["#0278d5"]);
+        }}
+        style={{
+          backgroundColor: color.includes("#0278d5") ? "#0278d5" : "",
+          width: "95px",
+        }}
```
Comment (darishodzic) [inline src/components/Navigation/index.tsx:31]: pls dont use inline css

File: src/components/Navigation/index.tsx [hunk 30]
```diff
@@ -0,0 +1,108 @@
...
+      <Text
+        onChange={handleChange}
+        onClick={() => {
+          navigate("/Flags");
+          setColor(["flags"]);
+        }}
+        style={{
+          backgroundColor: color.includes("flags") ? "#0278d5" : "",
+          width: "80px",
+        }}
```
Comment (darishodzic) [inline src/components/Navigation/index.tsx:44]: inline css

File: src/components/Navigation/index.tsx [hunk 31]
```diff
@@ -0,0 +1,108 @@
...
+      <Text
+        onClick={() => {
+          navigate("/Application");
+          setColor(["application"]);
+        }}
+        style={{
+          backgroundColor: color.includes("application") ? "#0278d5" : "",
+          width: "130px",
+        }}
+      >
```
Comment (darishodzic) [inline src/components/Navigation/index.tsx:69]: inline css

File: src/components/SignUp/index.tsx [hunk 32]
```diff
@@ -0,0 +1,112 @@
...
+        values.password === value ? null : "password do not match",
+    },
+  });
+  return (
+    <div
+      style={{
+        height: "100vh",
+        width: "100%",
+        backgroundImage: `url(${feature})`,
+      }}
```
Comment (darishodzic) [inline src/components/SignUp/index.tsx:47]: inline css

File: src/logo.svg [hunk 33]
```diff
@@ -0,0 +1,15 @@
...
+<path d="M1640 3770 c-96 -4 -251 -11 -345 -16 l-170 -8 135 -7 c368 -20 767 -29 1281 -29 l566 0 145 -93 c397 -257 499 -372 597 -677 41 -125 55 -224 48 -333 -14 -200 -78 -352 -210 -493 -187 -200 -404 -283 -707 -271 -351 14 -648 138 -905 378 -209 196 -409 508 -521 811 l-39 108 -82 0 c-72 0 -83 -2 -83 -17 0 -10 50 -364 110 -787 61 -423 110 -770 110 -772 0 -2 27 -4 60 -4 35 0 60 4 60 10 0 6 11 22 25 36 14 14 32 42 41 62 8 20 27 50 42 65 79 84 160 79 527 -27 371 -107 500 -133 705 -142 485 -23 918 133 1230 446 185 184 299 393 356 650 21 93 24 362 5 470 -35 204 -108 374 -231 538 l-32 42 416 0 416 0 -2 -797 -3 -798 -27 -57 c-68 -147 -238 -240 -461 -254 -45 -3 -97 -7 -114 -10 l-33 -5 0 -64 0 -65 1025 0 1026 0 -3 68 -3 67 -110 7 c-319 19 -499 139 -534 358 -7 41 -11 337 -11 808 l0 742 583 0 c332 0 737 7 942 15 198 8 383 15 410 16 117 2 6 9 -363 24 -329 13 -819 15 -3045 13 -1458 0 -2731 -4 -2827 -8z"/>
+<path d="M1716 1585 c4 -8 8 -15 10 -15 2 0 4 7 4 15 0 8 -4 15 -10 15 -5 0 -7 -7 -4 -15z"/>
+</g>
+</svg>
```
Comment (darishodzic) [inline src/logo.svg:15]: delete this pls

**Review by darishodzic — CHANGES_REQUESTED** [review]

**Pull request merged**